lib/fluent/plugin/in_tail.rb

```diff
@@ -911,16 +1011,34 @@ def opened?
         def handle_notify
           with_io do |io|
             begin
+              number_bytes_read = 0
+              start_reading = Fluent::Clock.now
               read_more = false
 
               if !io.nil? && @lines.empty?
                 begin
                   while true
-                    @fifo << io.readpartial(8192, @iobuf)
+                    @fifo << io.readpartial(BYTES_TO_READ, @iobuf)
                     @fifo.read_lines(@lines)
-                    if @lines.size >= @read_lines_limit
+
+                    number_bytes_read += BYTES_TO_READ
```

Comment:
It doesn't count actual read bytes correctly.
`BYTES_TO_READ` isn't actual read bytes, it maximum bytes to read.

Reply:
Previously, I'd tried to count actual bytes.

But the previous approach causes unintended behavior:

> I think that should be number_bytes_read += 8192, does not matter if it reached the end of the file at any read it try to read 8192 bytes.
> By using number_bytes_read += 8192, the throttling works as expected.

ref: https://github.com/fluent/fluentd/pull/3185#issuecomment-754545966

Reply:
> I think everything looks good except number_bytes_read += @lines.last ? @lines.last.size : 0.

It should be like this:
```diff
diff --git a/lib/fluent/plugin/in_tail.rb b/lib/fluent/plugin/in_tail.rb
index 2ca25494..ac8794ac 100644
--- a/lib/fluent/plugin/in_tail.rb
+++ b/lib/fluent/plugin/in_tail.rb
@@ -1018,10 +1018,11 @@ module Fluent::Plugin
               if !io.nil? && @lines.empty?
                 begin
                   while true
-                    @fifo << io.readpartial(BYTES_TO_READ, @iobuf)
+                    data = io.readpartial(BYTES_TO_READ, @iobuf)
+                    number_bytes_read += data.bytesize
+                    @fifo << data
                     @fifo.read_lines(@lines)
 
-                    number_bytes_read += BYTES_TO_READ
                     limit_bytes_per_second_reached = (number_bytes_read >= @read_bytes_limit_per_second && @read_bytes_limit_per_second > 0)
                     @log.debug("reading file: #{@path}")
                     if @lines.size >= @read_lines_limit || limit_bytes_per_second_reached
```
In the previous code you only count the last line, but incoming data may be like this
```
test
test
...
test
```

Reply:
Oh! I'd overlooked for `#bytesize`. :stuck_out_tongue:

Reply:
Fixed in https://github.com/fluent/fluentd/pull/3185/commits/92aadec7038204cc54cf241a21eb9e8fc0b5c573.